src/arc.ts

```diff
@@ -627,6 +629,17 @@ export class Arc extends GraphNodeObserver {
     Logger.debug(`Data saved successfully as ${descriptionHash}`)
     return descriptionHash
   }
+
+  public approveTokens(tokenAddress: Address, spender: Address, amount: BN) {
```

Comment:
who is calling this function ?and on which cases ?

Reply:
please add test for that.

Reply:
This is called on Alchemy when the user tries to create a proposal on TokenTrade Plugin. The use case is that the user needs to give approval to the TokenTrade Plugin instance of the DAO (before creating the proposal) so the tokens can be send to the DAO